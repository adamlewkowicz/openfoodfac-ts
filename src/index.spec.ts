import { OpenFoodFactsApi } from './index';
import openFoodFactsResponseMock from './__mocks__/response.json';

describe('OpenFoodFactsApi', () => {

  let openFoodFactsApi: OpenFoodFactsApi;

  beforeEach(() => openFoodFactsApi = new OpenFoodFactsApi());

  describe('findByBarcode()', () => {

    it('should normalize product correctly 📝', async () => {
      const barcode = '5900512300108';
      const fetchMock = fetch as jest.Mock;
    
      fetchMock.mockImplementationOnce(async () => ({
        ok: true,
        json: async () => openFoodFactsResponseMock
      }));
    
      const normalizedProduct = await openFoodFactsApi.findByBarcode(barcode);
    
      expect(fetchMock).toHaveBeenCalledTimes(1);
      expect(normalizedProduct).toMatchSnapshot();
    });

  });

});