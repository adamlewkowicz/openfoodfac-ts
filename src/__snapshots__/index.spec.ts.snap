// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenFoodFactsApi findByBarcode() should normalize product correctly üìù 1`] = `
Object {
  "_id": "5900512300108",
  "images": Array [
    "https://static.openfoodfacts.org/images/products/590/051/230/0108/front_en.3.400.jpg",
    "https://static.openfoodfacts.org/images/products/590/051/230/0108/nutrition_pl.22.400.jpg",
  ],
  "macro": Object {
    "carbs": 1,
    "fats": 82,
    "fattyAcids": 54,
    "kcal": 746,
    "prots": 1,
    "sugars": 1,
  },
  "name": "Mas≈Ço extra",
  "portion": 10,
  "portions": Array [
    Object {
      "type": "portion",
      "unit": "g",
      "value": 10,
    },
  ],
}
`;
